File: [U]-cpid_1107.py
# N = int(input())
name_map = {}
ZODIAZ = {'Ox':1, 'Tiger':2, 'Rabbit':3, 'Dragon':4, 'Snake':5, 'Horse':6, 'Goat':7, 'Monkey':8, 'Rooster':9, 'Dog':10, 'Pig':11, 'Rat':12}
# for i in range(N):
#     string = input().split(' ')
#     name = string[0]
#     relation = string[3]
#     animal = string[4]
#     name_map[name] = (relation, animal)

def calculate_between(pre_relation_tuple, relation_tuple):
    if relation_tuple[0] == 'previous':
        return abs(ZODIAZ[pre_relation_tuple[1]]-ZODIAZ[relation_tuple[1]])-12
    else:
        return abs(ZODIAZ[pre_relation_tuple[1]]-ZODIAZ[relation_tuple[1]])
print(calculate_between(('pre', 'Ox'), ('previous', 'Dragon')))

File: usaco/[U]-cpid_1107.cpp
#include <iostream>
#include <string>
#include <vector>
#include <map>

using namespace std;

int calculateBetween(pair<string, string>preRelationPair, pair<string, string>relationPair)
{
    map<string, int> ZODIAZ;
    ZODIAZ["Ox"] = 1; ZODIAZ["Tiger"] = 2; ZODIAZ["Rabbit"] = 3; ZODIAZ["Dragon"] = 4; ZODIAZ["Snake"] = 5; ZODIAZ["Horse"] = 6; ZODIAZ["Goat"] = 7; ZODIAZ["Monkey"] = 8; ZODIAZ["Rooster"] = 9; ZODIAZ["Dog"] = 10; ZODIAZ["Pig"] = 11; ZODIAZ["Rat"] = 12;
    if (relationPair.first == "previous")
    {
        return abs(ZODIAZ[preRelationPair.second]-ZODIAZ[relationPair.second])-12;
    } else
    {
        return abs(ZODIAZ[preRelationPair.second]-ZODIAZ[relationPair.second]);
    }
}

int main()
{
    cout << calculateBetween(pair<string, string> ("pre", "Ox"), pair<string, string>("previous", "Dragon")) << endl;
}